core/src/main/java/hudson/util/StreamTaskListener.java

```diff
@@ -58,7 +52,7 @@
  * 
  * @author Kohsuke Kawaguchi
  */
-public class StreamTaskListener extends AbstractTaskListener implements Serializable, Closeable {
+public class StreamTaskListener implements TaskListener, Closeable {
```

Comment:
🐛  You can no longer cast to AbstractTaskListener or check for instanceof AbstractTaskListener.

Reply:
That is true, but `AbstractTaskListener` was only ever a convenience base class, so there was never a legitimate reason to do so to begin with.

Reply:
I was unable to find the implementation, which may be affected. But I agree that keeping the  inheritance from a deprecated class would be the safest option for now. This old class is not doing anything wrong anyway

Reply:
As in #2879 I would prefer to eliminate all usages of the deprecated type to reduce confusion, since there is no plausible risk from the technical API change here.